### PR TITLE
RAC-6079: Refactor vagrant cache build shell script

### PR DESCRIPTION
**Pull request by PengTian0**
Implement a shell script build_ovf.sh for vagrant cache build pipeline.
@panpan0000 @changev @anhou @iceiilin

### CODE REVIEW

File: packer/box_handler.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        echo "[ERROR] packer is not installed . please install it."
+        exit 5
+    fi
+    vagrant_path=$( which vagrant )
+    if [ -z "$vagrant_path" ];then
+        echo "[ERROR] vagrant is not installed . please install it."
+        exit 5
+    fi
+    jq_path=$(which jq)
+    if [ ! -n "$(which jq)" ] ; then
```
Comment (panpan0000) [inline packer/box_handler.sh]: are you going to use   ```if [ -z $jq_path ] ; then``` like what you did before?

File: packer/build_ovf.sh [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    echo "      --TARGET_DIR: The target directory to put ovf and log file"
+}
+
+#########################################
+#
+#  build ovf
+#
+#########################################
+build(){
+    pushd $DIR
```
Comment (panpan0000) [inline packer/build_ovf.sh]: do you want to use $1/$2/$3 instead of global variable ..?

File: packer/build_ovf.sh [hunk 3]
```diff
@@ -0,0 +1,113 @@
+#!/bin/bash -e
```
Comment (panpan0000) [inline packer/build_ovf.sh]: ```ovf``` is a standard packaging format . it's not only used for Virtualbox/Vagrant , but also for VMWare( but I believe you will have another script to build the cached/templated RackHD Vmware image )
 I think it's more proper to rename this script to build_base_vagrant_box.sh or something more clear ?

File: packer/box_handler.sh (hunk 1 above)
Comment (PengTian0) [reply packer/box_handler.sh]: Thanks, fixed.

**Comment by PengTian0** [comment]
@panpan0000 Thanks.
Already rename the build_ovf.sh to "build_vagrant_base_box.sh" according to your comments.

**Pull request merged**